File: app/src/main/java/com/example/pizzeria/clases/Usuario.java
package com.example.pizzeria.clases;

import android.os.Build;

import java.io.Serializable;
import java.nio.charset.StandardCharsets;
import java.util.Base64;
import java.util.Objects;

public class Usuario implements Serializable {

    private String nombreUsuario;
    private String contrasenna;

    public Usuario(String nomUsuario, String contrasenya) {
        this.nombreUsuario = nomUsuario;
        this.contrasenna = contrasenya;
    }

    public void setNombreUsuario(String nombreUsuario) {
        this.nombreUsuario = nombreUsuario;
    }

    public String getNombreUsuario() {
        return this.nombreUsuario;
    }

    public void setContrasenna(String contrasenna) { this.contrasenna = Contrasenna.Companion.cifrar(contrasenna);}

    public String getContrasenna() {
        return this.contrasenna;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Usuario usuario = (Usuario) o;
        return nombreUsuario.equals(usuario.nombreUsuario);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nombreUsuario);
    }

    @Override
    public String toString() {
        return "Usuario(nombreUsuario='" + nombreUsuario + "', contrasenna='" + contrasenna + "')";
    }
}

File: app/src/main/java/com/example/pizzeria/clases/Usuario.kt
package com.example.pizzeria.clases

import android.os.Build
import java.io.Serializable
import java.util.Base64

class Usuario (private var nombreUsuario: String, contrasenna: String): Serializable {

    private var contrasenna: String

    init {
        this.contrasenna = cifrar(contrasenna)
    }

    private fun cifrar(contrasenna: String): String {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    Base64.getEncoder().encodeToString(contrasenna.toByteArray(Charsets.UTF_8))
                } else {
                    contrasenna
                }
    }

    fun getNombreUsuario(): String {
        return this.nombreUsuario
    }

    fun setNombreUsuario(nombreUsuario: String) {
        this.nombreUsuario = nombreUsuario
    }

    fun getContrasenna(): String {
        return this.contrasenna
    }

    fun setContrasenna(contrasenna: String) {
        this.contrasenna = cifrar(contrasenna)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Usuario

        if (nombreUsuario != other.nombreUsuario) return false

        return true
    }

    override fun hashCode(): Int {
        return nombreUsuario.hashCode()
    }

    override fun toString(): String {
        return "Usuario(nombreUsuario='$nombreUsuario', contrasenna='$contrasenna')"
    }

}